app/plugin_api.go

```diff
@@ -311,6 +311,21 @@ func (api *PluginAPI) UpdateUserStatus(userId, status string) (*model.Status, *m
 	return api.app.GetStatus(userId)
 }
 
+func (api *PluginAPI) UpdateUserStatusV2(userId, status, endtime string) (*model.Status, *model.AppError) {
+	switch status {
+	case model.STATUS_ONLINE:
+	case model.STATUS_OFFLINE:
+	case model.STATUS_AWAY:
+		api.UpdateUserStatus(userId, status)
+	case model.STATUS_DND:
+		api.app.SetStatusDoNotDisturbTimed(userId, endtime)
+	default:
+		return nil, model.NewAppError("UpdateUserStatusV2", "plugin.api.update_user_status.bad_status", nil, "unrecognized status", http.StatusBadRequest)
+	}
+
+	return api.app.GetStatus(userId)
```

Comment:
We can simplify it a lot with something like:
```suggestion
	if status = model.STATUS_DND {
		api.app.SetStatusDoNotDisturbTimed(userId, endtime)
		return api.app.GetStatus(userId)
	}
	return api.UpdateUserStatus(userId, status)
```